lms/djangoapps/teams/csv.py

```diff
@@ -306,6 +314,41 @@ def validate_user_assignment_to_team_and_teamset(self, row):
                 return False
         return True
 
+    def validate_compatible_enrollment_modes(self, user, team_name, teamset_id):
+        """
+        Validates that only students enrolled in a masters track are on a single team. Disallows mixing of masters
+        with other enrollment modes on a single team.
+        """
+        if(teamset_id, team_name) not in self.user_enrollment_by_team:
+            self.user_enrollment_by_team[teamset_id, team_name] = set()
+        self.user_enrollment_by_team[teamset_id, team_name].add(self.user_to_actual_enrollment_mode[user.id])
+        if self.is_FERPA_bubble_breached(teamset_id, team_name):
+            error_message = \
+                'Team {} cannot have Master’s track users mixed with users in other tracks.'.format(team_name)
+            self.add_error_and_check_if_max_exceeded(error_message)
+            return False
+        return True
+
+    def is_FERPA_bubble_breached(self, teamset_id, team_name):
+        """
+        Ensures that FERPA bubble is not breached.
+        Checks that we are not trying to violate FERPA proctection by mixing masters
+        track students with other enrollment tracks.
+        """
+
+        team_enrollment_modes = self.user_enrollment_by_team[teamset_id, team_name]
+        protected_modes = set(ORGANIZATION_PROTECTED_MODES)
+
+        if team_enrollment_modes.isdisjoint(protected_modes):
+            return False
+        elif team_enrollment_modes == protected_modes:
+            return False
+        elif team_enrollment_modes.issubset(protected_modes):
```

Comment:
we you don't need both of these, i don't think, the issubset should suffice i think. Also, i don't think you need the return at the end if you're using the else, i'd either have the exterior return or the else but not both